src/System.IO.Pipelines.Networking.Tls/Internal/OpenSsl/LockStore.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines.Networking.Tls.Internal.OpenSsl
+{
+    internal unsafe static class LockStore
+    {
+        private static SemaphoreSlim[] _locks;
+        public static readonly InteropCrypto.locking_function Callback;
```

Comment:
Agreed, I also dislike. The class was public, but I realised it didn't need to be and wasn't updated. Now changed.